benchmarks/profiler_gc.rb

```diff
@@ -0,0 +1,154 @@
+
+  def run_benchmark
+    Benchmark.ips do |x|
+      benchmark_time = VALIDATE_BENCHMARK_MODE ? { time: 0.01, warmup: 0 } : { time: 10, warmup: 2 }
+      x.config(
+        **benchmark_time,
+        suite: report_to_dogstatsd_if_enabled_via_environment_variable(benchmark_name: 'profiler_gc')
+      )
+
+      # The idea of this benchmark is to test the overall cost of the Ruby VM calling these methods on every GC.
+      # We're going as fast as possible (not realistic), but this should give us an upper bound for expected performance.
+      x.report('profiler gc') do
+        Datadog::Profiling::Collectors::ThreadContext::Testing._native_on_gc_start(@collector)
+        Datadog::Profiling::Collectors::ThreadContext::Testing._native_on_gc_finish(@collector)
+        Datadog::Profiling::Collectors::ThreadContext::Testing._native_sample_after_gc(@collector)
+      end
+
+      x.save! 'profiler-gc-results.json' unless VALIDATE_BENCHMARK_MODE
+      x.compare!
+    end
+
+    Benchmark.ips do |x|
+      benchmark_time = VALIDATE_BENCHMARK_MODE ? { time: 0.01, warmup: 0 } : { time: 10, warmup: 2 }
+      x.config(
+        **benchmark_time,
+        suite: report_to_dogstatsd_if_enabled_via_environment_variable(benchmark_name: 'profiler_gc_minute')
+      )
+
+      # We cap the number of minor GC samples to not happen more often than TIME_BETWEEN_GC_EVENTS_NS (10)
+      minor_gc_per_second_upper_bound = 100
+      # ...but every major GC triggers a flush. Here we consider what would happen if we had 1000 major GCs per second
+      pessimistic_number_of_gcs_per_second = minor_gc_per_second_upper_bound * 10
+      estimated_gc_per_minute = pessimistic_number_of_gcs_per_second * 60
+
+      x.report("estimated profiler gc per minute (sample #{estimated_gc_per_minute} times + serialize result)") do
+        estimated_gc_per_minute.times do
+          Datadog::Profiling::Collectors::ThreadContext::Testing._native_on_gc_start(@collector)
+          Datadog::Profiling::Collectors::ThreadContext::Testing._native_on_gc_finish(@collector)
+          Datadog::Profiling::Collectors::ThreadContext::Testing._native_sample_after_gc(@collector)
+        end
+
+        @recorder.serialize
+      end
+
+      x.save! 'profiler-gc-minute-results.json' unless VALIDATE_BENCHMARK_MODE
+      x.compare!
+    end
+
+    Benchmark.ips do |x|
+      benchmark_time = VALIDATE_BENCHMARK_MODE ? { time: 0.01, warmup: 0 } : { time: 10, warmup: 2 }
+      x.config(
+        **benchmark_time,
+        suite: report_to_dogstatsd_if_enabled_via_environment_variable(benchmark_name: 'profiler_gc_integration')
+      )
+
+      x.report('Major GC runs (profiling disabled)', 'GC.start')
+
+      x.save! 'profiler-gc-integration-results.json' unless VALIDATE_BENCHMARK_MODE
+      x.compare!
+    end
+
+    Datadog.configure do |c|
+      c.profiling.enabled = true
+      c.profiling.allocation_enabled = false
+      c.profiling.advanced.force_enable_gc_profiling = true
+    end
```

Comment:
Yes! I decided to go ahead and add a helper for that, I wanted to avoid copy-pasting the `_native_is_running?` loop more times ;)

Fixed in bfbf049fd00755f3d3faa69795dc7513413e0ff4